crates/ide-assists/src/utils.rs

```diff
@@ -250,6 +252,35 @@ pub fn add_trait_assoc_items_to_impl(
     first_item.unwrap()
 }
 
+pub(crate) fn indent_string(mut s: &str, indent_level: IndentLevel) -> Cow<'_, str> {
+    if indent_level.is_zero() || s.is_empty() {
+        return s.into();
+    }
+    let level = indent_level.0 as usize;
+
+    let indent = indent_level.to_string();
+    let mut buf = String::with_capacity(s.len() + level * 3 * 4);
+
+    if !s.starts_with('\n') {
+        buf.push_str(&indent);
+    }
+
+    while let Some((line, rest)) = s.split_once('\n') {
+        buf.push_str(line);
+        buf.push('\n');
+
+        if !rest.split_once('\n').map_or(rest, |s| s.0).is_empty() {
```

Comment:
can you pull out the `split_once`/`map_or` into a separate statement? it makes it a bit tricky to understand what the condition here _actually is_.